### DISCUSSION TITLE
[Resolved] Short. vs Long `linkTitle`

### DISCUSSION THREAD

**Discussion by svrnm** [open-telemetry/opentelemetry.io#5039] [Information Architecture]
Taking this discussion out of https://github.com/open-telemetry/opentelemetry.io/pull/4966 and https://github.com/open-telemetry/opentelemetry.io/discussions/4853:

To make the new proposed information architecture successful, we need to consider to rethink our preference for short (or very short) linkTitle, which will be used in the side navigation presented to end users. One example among a few is "SDK" vs "Managing telemetry with the SDK" (see https://github.com/open-telemetry/opentelemetry.io/pull/4966#discussion_r1712261442).

I expressed my concerns [in this comment](https://github.com/open-telemetry/opentelemetry.io/discussions/4853#discussioncomment-10212657): 

> My only issue with the structure is that for a person not involved in OpenTelemetry (or observability) the structure will not help them to navigate, since they might know what an API & SDK are, but not know immediately what the difference is.

So if we go with "SDK" we have to assume that people read some of the introductions to understand what to expected when they click on it. My assumption is always that people read nothing and just look for the one thing they currently care about (e.g. "managing telemetry" or "capture telemetry"), so they will not find that if the title is "SDK" and will tell that our documentation sucks.

I'd like to understand from @open-telemetry/docs-maintainers, @open-telemetry/docs-approvers and others, what your preference is and what advantages you see between the one or the other? 

We need to make a decision eventually, and then write this down in our contribution guidelines.

**Comment by theletterf** [comment]
Thanks for raising this. I'd rather go with long linkTitles (though within reason). Why?

Because...

- They are clearer for all, including localizers and users with impaired vision using screen readers.
- We shouldn't be slaves of the standard desktop layout. In mobile views, the question doesn't make sense:

<img width="559" alt="Screenshot 2024-08-14 at 12 17 18" src="https://github.com/user-attachments/assets/089e0753-73da-4411-96ef-a88dd18060a7">

Kubernets is following that path already, it seems: https://kubernetes.io/docs/setup/production-environment/tools/

**Reply by cartermp** [reply]
I agree with this, especially since we can't assume people will read various implied prerequisite docs up front. The reality is they're going to hunt and peck for the specific things they need at the moment, and longer linktitles help with that.

**Reply by jaydeluca** [reply]
I also agree. I prefer clarity and context over conciseness

**Reply by tiffany76** [reply]
I agree as well. Favoring practicality over aesthetics makes sense to me, given how broad the documentation is.

**Reply by chalin** [reply]
Works for me, but at some point we'll want to keep some **balance**. With too many line wraps due to long titles in the left nav, it'll be difficult to parse / read.

**Reply by svrnm** [reply]
Yes, we need to keep a balance here. Can we _enforce_ a character limit somehow? Maybe some of our linters can help with that? We can still decide to raise that limit from time to time, but at least we keep track of it.

**Reply by theletterf** [reply]
I think 50 characters is a good limit — it'd translate to a double-line menu item in most desktop resolutions. Below is a 37 char example borrowed from Noam Chomsky.

<img width="835" alt="Screenshot 2024-09-04 at 09 54 01" src="https://github.com/user-attachments/assets/a40a0035-1069-4e4b-90c3-853eadc3d891">

**Reply by svrnm** [reply]
Yes, something between 40 and 50 was also what I was thinking about.

**Reply by chalin** [reply]
Let's track this idea via:

- #5152

**Comment by svrnm** [comment]
Since we all agree that we are OK with longer titles to accomplish more clarity, I will close this discussion now. If we ever need to revisit this decision, let's have a new discussion
